.github/workflows/review_and_testing.yml

```diff
@@ -1,10 +1,33 @@
-name: testing
+name: Review & Test
+
+permissions:
+  contents: read
+  pull-requests: write
 
 on:
   pull_request:
     branches: [main]
+    
+concurrency:
+  group:
+    ${{ github.repository }}-${{ github.event.number || github.head_ref ||
+    github.sha }}-${{ github.workflow }}-${{ github.event_name ==
+    'pull_request_review_comment' && 'pr_comment' || 'pr' }}
+  cancel-in-progress: ${{ github.event_name != 'pull_request_review_comment' }}
 
 jobs:
+  Review:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: coderabbitai/ai-pr-reviewer@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GT_TOKEN }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        with:
+          debug: false
+          review_simple_changes: false
+          review_comment_lgtm: false
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The "Review" job uses the `coderabbitai/ai-pr-reviewer` action for AI-based code reviews. It's important to note that the `GITHUB_TOKEN` and `OPENAI_API_KEY` secrets are required for this action to work. Make sure these secrets are properly set in your repository settings.

```diff
  jobs:
    Review:
      runs-on: ubuntu-latest
      steps:
        - uses: coderabbitai/ai-pr-reviewer@latest
          env:
            GITHUB_TOKEN: ${{ secrets.GT_TOKEN }}
            OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
          with:
            debug: false
            review_simple_changes: false
            review_comment_lgtm: false
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->